PlanningPoker.Tests/ServicesTests.cs

```diff
@@ -5,108 +5,155 @@
+        private readonly Room _room = new Room { Id = One, Name = One, CreatorName = One };
+
+        private readonly User _user = new User
+        {
+            RoomId = One,
+            Name = One,
+            Id = 1,
+            Vote = 1,
+            ConnectionId = One,
+            Email = One,
+            Password = One
+        };
+
         public ServicesTests()
         {
-            var hubContext = Substitute.For<IHubContext<LoopyHub>>();
-            _sut = new UserService(hubContext);
+            _userRepository = Substitute.For<IUserRepository>();
+            _roomsRepository = Substitute.For<IRoomsRepository>();
+            _hubContext = Substitute.For<IHubContext<LoopyHub>>();
+            var unitOfWork = Substitute.For<IUnitOfWork>();
+            _sutForUserService = new UserService(_hubContext, unitOfWork, _roomsRepository, _userRepository);
+            _sutForRoomService = new RoomService(_hubContext, unitOfWork, _roomsRepository, _userRepository);
         }
 
 
         [Fact]
-        public void GetRooms_Shlould_Return_Romm_If_Room_Was_Added()
+        public async Task RoomRepository_AddAsync_Should_Calls_And_Hubcontext_Should_Send_AddRoom_When_AddRoom_Called()
+        {
+            // Act
+            await _sutForRoomService.AddRoom(_room);
+            // Assert
+            await _roomsRepository.Received(1).AddAsync(Arg.Any<Room>());
+            await _hubContext.Received(1).Clients.All.SendAsync(Arg.Any<string>());
+        }
+        [Fact]
+        public void UserRepository_AddAsync_Should_Calls_When_AddUser_Called()
         {
-            // Arrange
             // Act
-            _sut.AddRoom(_room);
-            var result = _sut.GetRooms();
+            _sutForUserService.AddUser(_user);
 
             // Assert
-            Assert.Equal(result.First(), _room);
+            _userRepository.Received(1).AddAsync(Arg.Any<User>());
         }
         [Fact]
-        public void GetUserByConnection_Should_return_UserName_If_User_was_added()
+        public void UserRepository_UpdateAsync_Should_Calls_When_AddUserConnection_Called()
         {
-            // Arrange
+            //Arrange
+            _userRepository.GetByNameAsync(One).Returns(_user);
             // Act
-            _sut.AddUserConnection(_userConnection.ConnectionId, _userConnection.Name);
-            var result = _sut.GetUserByConnection(_userConnection.ConnectionId);
+            _sutForUserService.AddUserConnection(One, One, One);
             // Assert
-            Assert.Equal(result, _userConnection.Name);
+            _userRepository.Received(1).GetByNameAsync(Arg.Any<string>());
+            _userRepository.Received(1).Update(Arg.Any<User>());
         }
         [Fact]
-        public void GetVotesForRoom_should_return_UserVote_If_Votes_Were_added()
+        public void UserRepository_UpdateAsync_Should_Calls_And_Hubcontext_Should_Send_Vote_When_AddVote_Called()
```

Comment:
https://dzone.com/articles/7-popular-unit-test-naming I would suggest to choose one of them and use it. Because UserRepository at the begging of the test name confuses me